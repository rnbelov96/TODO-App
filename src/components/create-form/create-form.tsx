import { AppActionCreators } from '@/redux/app';
import { FormPropsType } from '@/types/components/form';
import { FullStateType } from '@/types/general-types';
import { AppActionType } from '@/types/redux/app-reducer';
import React from 'react';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

export const PureCreateForm: React.FunctionComponent<FormPropsType> = ({
  inputText,
  onSubmit,
  onChange,
}) => (
  <form
    className="form"
    onSubmit={e => {
      e.preventDefault();
      onSubmit();
    }}
  >
    <input
      placeholder="What needs to be done?"
      onChange={e => {
        onChange(e.target.value);
      }}
      value={inputText}
      type="text"
    />
  </form>
);

function mapStateToProps(state: FullStateType) {
  return {
    inputText: state.app.inputText,
  };
}

function mapDispatchToProps(dispatch: Dispatch<AppActionType>) {
  return {
    onChange: (data: string) => dispatch(AppActionCreators.changeInput(data)),
    onSubmit: () => dispatch(AppActionCreators.submit()),
  };
}

export const CreateForm = connect(mapStateToProps, mapDispatchToProps)(PureCreateForm);
