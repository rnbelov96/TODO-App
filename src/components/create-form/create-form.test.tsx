import React from 'react';
import renderer from 'react-test-renderer';
import { PureCreateForm } from './create-form';

describe('CreateForm', () => {
  test('should render create form', () => {
    const tree = renderer
      .create(
        <PureCreateForm
          inputText="Some text"
          onSubmit={() => {}}
          onChange={() => {}}
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
