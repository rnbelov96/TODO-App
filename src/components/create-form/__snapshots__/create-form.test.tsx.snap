// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateForm should render create form 1`] = `
<form
  className="form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="What needs to be done?"
    type="text"
    value="Some text"
  />
</form>
`;
